File: 704-Binary Search-二分查找/704-Binary Search.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        int mid = (left + right) / 2;
        while(left <= right) {
        	if(nums[mid] == target) {
        		return mid;
        	}else if(target > nums[mid]) {
        		left = mid + 1;
        	}else {
        		right = mid - 1;
        	}
        	mid = (left + right) / 2;
        }
        return -1;
    }
}

File: 704-Binary Search-二分查找/704-Binary Search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
 var search = function(nums, target) {
    return biSearch(nums, target, 0, nums.length - 1)
};

const biSearch = function(nums, target, left, right){
    if(left > right) return -1
    const mid = Math.floor((left + right) / 2)
    if(nums[mid] == target){
        return mid
    }else if(nums[mid] > target){
        return biSearch(nums, target, left, mid - 1)
    }else{
        return biSearch(nums, target, mid + 1, right)
    }
}